README.md

```diff
@@ -1,7 +1,170 @@
+
+        // This will be the input to our policy.
+        java.util.Map<String,Object> input = java.util.Map.ofEntries(
+            entry("subject", "alice"),
+            entry("action", "read"),
+            entry("resource", "/finance/reports/fy2038_budget.csv")
+        );
+        System.out.println("input for OPA request is: " + input);
+
+        // We will read the list of policy violations, and whether the request
+        // is allowed or not into these.
+        java.util.List<Object> violations;
+        boolean allowed;
+
+        // Perform the request against OPA.
+        try {
+            allowed = opa.check(input, "policy/allow");
+            violations = opa.query(input, "policy/violations");
+        } catch (OPAException e ) {
+            // Note that OPAException usually wraps other exception types, in
+            // case you need to do more complex error handling.
+            System.out.println("exception while making request against OPA: " + e);
+            throw e; // crash the program
+        }
+
+        System.out.println("allowed: " + allowed);
+        System.out.println("violations: " + violations);
+    }
+}
+```
+
+Before we can run this program, we also need a policy, and an OPA instance running. Create the file `main.rego` with these contents:
+
+```rego
+package policy
+
+import rego.v1
+
+groups := {
+    "accounting": {
+        "members": [
+            "alice", "bob",
+        ]
+    },
+    "it": {
+        "members": ["eve"]
+    }
+}
+
+subject := object.get(input, "subject", "")
+
+groups_for_subject contains group if {
+    groups[group].members[_] == subject
+}
+
+violations contains violation if {
+    count(groups_for_subject) == 0
+    violation := sprintf("subject '%s' is not associated with any groups", [subject])
+}
+
+violations contains violation if {
+    startswith(input.resource, "/finance")
+    not "accounting" in groups_for_subject
+    violation := sprintf("subject '%s' s not in the accounting group", [subject])
+}
+
+default allow = false
+
+allow = true if {
+    count(violations) == 0
+}
+```
+
+Now run `opa run -s main.rego`. Leave this command running while you run your Java program.
+
+Finally, we can run the program with `./gradlew run` and see it communicating with the Rego policy:
+
+```plain
+$ ./gradlew run
+
+> Task :app:run
+Hello World!
+input for OPA request is: {subject=alice, resource=/finance/reports/fy2038_budget.csv, action=read}
+allowed: true
+violations: []
+
+BUILD SUCCESSFUL in 1s
+2 actionable tasks: 1 executed, 1 up-to-date
+```
+
+Try changing the policy input in `App.java`, or modifying the `main.rego` to observe how the behavior will change. Remember that you will need to restart `opa` any time you edit `main.rego`.
+
+## License
+
 [![License](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 
+
+
+*content below this lines is managed by Speakeasy's automated generation tools*
```

Comment:
Hmm. I'd prefer if we found a better way. I don't really want us to disclaim ownership of the docs like that. If the generated docs are ~trash~ insufficient, that's our problem to deal with, we can't say that it's SE's generated docs and be done with it.

We can remote sections:

> If you would like to take over the management of automatically generated sections, then you can do the following:
> 
>   1.  Remove the <!-- Start ... -> section comment.
>   2.  Find the matching <!-- End ... --> comment and change it to <!-- No ... -->, which marks that section as managed by you. (This step is important. If you only remove the "Start" comment, the section may be re-inserted as described below.)
>   3.  Edit the content that was between those comments as you see fit.
>

Reply:
Yeah that’s a fair point. TBH I should probably just integrate some usage example of the SE generated API and any other pertinent info in the generated readme, then remove the generated sections. 

Should probably make some tests that will trip if the API changes too.

Reply:
I went ahead and adopted the exact same way you had it.